### DISCUSSION TITLE
I failed to run `make pre-commit -B`, error `Unable to generate protoc`

### DISCUSSION THREAD

**Discussion by JessieTeng89** [argoproj/argo-workflows#8865] [Q&A] [answered]
Hi guys,

 I tried to follow the guide https://argoproj.github.io/argo-workflows/running-locally/ to run `make pre-commit -B`, but I got below error, any idea?

```
2022/05/26 18:22:40 Unable to generate protoc on k8s.io.apimachinery.pkg.util.intstr: exit status 1
make: *** [pkg/apis/workflow/v1alpha1/generated.proto] Error 1
```

**Comment by ibotdotout** [comment, marked as answer]
I faced this issue and solved by

- move repo to `~/go/src/github.com/argoproj/argo-workflows` as mentioned in [this contribution guide](https://www.youtube.com/watch?v=zZv0lNCDG9w&t=1255s)

```
⚠️ Code generation will not work if code in not checked out into /src/github.com/argoproj/argo-workflows
```

- cleanup by `make clean`
- run `make pre-commit -B` and good luck :)

References:  
- https://github.com/argoproj/argo-workflows/issues/7471#issuecomment-1004263109
- https://github.com/argoproj/argo-workflows/pull/9235#issuecomment-1196820203

**Comment by ebuildy** [comment]
What platform (linux / macos ...) ?
